include/base/dof_map.h

```diff
@@ -1616,7 +1634,7 @@ class DofMap : public ReferenceCountedObject<DofMap>,
    * constraints or sufficiently many user constraints.
    */
   std::unique_ptr<SparsityPattern::Build> build_sparsity(const MeshBase & mesh,
-                                                         bool calculate_constrained = false) const;
+                                                         const bool calculate_constrained = false) const;
```

Comment:
At some point we've got to make a global decision about when pass-by-value parameters should be `const`.  On the one hand it's helpful for protection the implementation, but on the other hand it just muddies up the declaration with an implementation-specific detail that doesn't actually promise anything.

Reply:
Yeah, since we don't pass `const` non-reference parameters like this consistently elsewhere in the library, I agree it looks a bit out of place to make this change. I could be talked into inconsistently having `const` parameters in the .C file based on the author's preference, but I'm not convinced that really helps catch lots of bugs either.